raiden/raiden_service.py

```diff
@@ -258,30 +256,34 @@ def start(self):
                 block_number,
             )
             self.wal.log_and_dispatch(state_change, block_number)
+
+            # On first run Raiden needs to fetch all events for the payment
+            # network, to reconstruct all token network graphs and find opened
+            # channels
+            last_log_block_number = None
         else:
-            # Get the last known block number after reapplying all the state changes from the log
+            # The `Block` state change is dispatch only after all the events
+            # for that given block have been processed, filters can be safely
+            # installed starting from this position without losing events.
             last_log_block_number = views.block_number(self.wal.state_manager.current_state)
 
         # The alarm task must be started after the snapshot is loaded or the
-        # state is primed, the callbacks assume the node is initialized.
-        self.alarm.start()
+        # state is primed, otherwise the state changes won't have effect.
```

Comment:
Actually, this is not really correct. It doesn't really matter when the alarm task is installed, what matters is that we install the event filters after the state is primed.